src/transformers/pipelines/token_classification.py

```diff
@@ -228,6 +234,70 @@ def __call__(self, inputs: Union[str, List[str]], **kwargs):
             return answers[0]
         return answers
 
+    def set_subwords_label(self, entities: List[dict]) -> List[dict]:
+        strategy = self.subword_label_re_alignment
+
+        def set_labels(sub_words: List[dict]) -> dict:
+            score = np.stack([sub["score"] for sub in sub_words])
+            if strategy == "default" or strategy is True:
+                label_idx = score[0].argmax()
+                label = self.model.config.id2label[label_idx]
+                sub_words[0]["entity"] = label
+                sub_words[0]["score"] = score[0][label_idx]
+                for sub in sub_words[1:]:
+                    sub["entity"] = "O"
+                    sub["score"] = 0.0  # what score should we assign here?
+            else:
+                if strategy == "first":
+                    # get label of first sub-word
+                    max_label_idx = score[0].argmax()
+                    label = self.model.config.id2label[max_label_idx]
+                elif strategy == "max":
+                    max_label_idx = np.unravel_index(np.argmax(score, axis=None), score.shape)[1]
+                    label = self.model.config.id2label[max_label_idx]
+                elif strategy == "average":
+                    max_label_idx = np.mean(score, axis=0).argmax()
+                    label = self.model.config.id2label[max_label_idx]
+                else:
+                    raise ValueError(f"Invalid value {strategy} for option `subword_label_re_alignment`")
```

Comment:
This should be raised earlier, preferably at the very beginning (though if we go with an Enum, it will be automatic).